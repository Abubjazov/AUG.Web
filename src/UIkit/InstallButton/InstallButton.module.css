.root {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 26px;
  font-size: 12px;
  font-weight: 500;
  line-height: 14px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  border: 0;
  border-radius: 50px;
  transition: all var(--transition-delay) linear;
}

.desktop {
  padding: 6px 20px;
}

.mobile {
  width: 26px;
  padding: 6px;
}

.install {
  color: var(--white);
  background-color: var(--blue);
}

.installed {
  color: var(--white);
  background-color: var(--gray);
}

.uninstall {
  color: var(--white);
  background-color: var(--red);
}

.disabled {
  cursor: not-allowed;
}

.display-none {
  display: none;
}

@media (hover: hover) {
  .install:hover {
    color: var(--blue);
    background-color: var(--white-o);
  }

  .install:hover svg path {
    stroke: var(--blue);
  }
}
