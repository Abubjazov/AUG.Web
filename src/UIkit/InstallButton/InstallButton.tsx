import { FC, useState } from 'react'

import { installDapplet, unInstallDapplet } from 'services/userData/userData'
import { useAppDispatch, useAppSelector } from 'store/hooks'
import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './InstallButton.module.css'

export enum InstallButtonMode {
  INSTALL = 'install',
  INSTALLED = 'installed',
  UNINSTALL = 'uninstall',
  DISPLAY_NONE = 'display-none',
}

export interface InstallButtonProps {
  dappletId: string
  loading?: boolean
  disabled?: boolean
  mobile?: boolean
  setMode?: InstallButtonMode
}

const InstallButton: FC<InstallButtonProps> = ({
  dappletId,
  loading = false,
  disabled = false,
  mobile = false,
  setMode,
}) => {
  const [unInstallMode, setUnInstallMode] = useState(false)
  const dispatch = useAppDispatch()

  const myDapplets = useAppSelector(state => state.userData.userDapplets)

  const targetMyDapplets = myDapplets.filter(
    dapplet => dapplet.dappletId === dappletId,
  )[0]

  const modeSetter = (setMode?: InstallButtonMode, mobile?: boolean) => {
    if (mobile && !setMode) {
      return targetMyDapplets && targetMyDapplets?.dappletState
        ? InstallButtonMode.INSTALLED
        : InstallButtonMode.INSTALL
    }

    if (mobile && setMode) {
      return targetMyDapplets?.dappletState
        ? setMode
        : InstallButtonMode.DISPLAY_NONE
    }

    return targetMyDapplets && targetMyDapplets?.dappletState
      ? unInstallMode
        ? InstallButtonMode.UNINSTALL
        : InstallButtonMode.INSTALLED
      : InstallButtonMode.INSTALL
  }

  const mode = modeSetter(setMode, mobile)

  const onClick = (e: { stopPropagation: () => void }) => {
    e.stopPropagation()

    if (mode === InstallButtonMode.INSTALL)
      void dispatch(installDapplet(dappletId))

    if (mode === InstallButtonMode.UNINSTALL)
      void dispatch(unInstallDapplet(dappletId))
  }

  return loading ? (
    <div
      className={cc([
        styles.root,
        `${mobile ? styles['skeleton-mobile'] : styles['skeleton-desktop']}`,
      ])}
    ></div>
  ) : (
    <button
      type="button"
      data-testid="install-button"
      disabled={disabled}
      className={cc([
        styles.root,
        `${mobile ? styles.mobile : styles.desktop}`,
        styles[mode],
        `${disabled ? styles.disabled : ''}`,
      ])}
      onClick={onClick}
      onMouseEnter={() => setUnInstallMode(true)}
      onMouseLeave={() => setUnInstallMode(false)}
    >
      {mobile ? <SvgIcon icon={mode} /> : mode}
    </button>
  )
}

export default InstallButton
