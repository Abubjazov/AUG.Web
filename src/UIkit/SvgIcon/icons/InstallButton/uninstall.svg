<svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.00008 12.5681C10.1782 12.5681 12.7546 10.0752 12.7546 6.99994C12.7546 3.92472 10.1782 1.43176 7.00008 1.43176C3.82197 1.43176 1.24561 3.92472 1.24561 6.99994C1.24561 10.0752 3.82197 12.5681 7.00008 12.5681Z" stroke="white" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M8.72612 5.32959L5.27344 8.6705" stroke="white" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M5.27344 5.32959L8.72612 8.6705" stroke="white" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
