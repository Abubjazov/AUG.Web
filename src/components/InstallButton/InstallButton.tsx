import { FC } from 'react'

import SvgIcon from 'components/SvgIcon'

import styles from './InstallButton.module.css'

export enum ButtonMode {
  INSTALL = 'install',
  INSTALLED = 'installed',
  UNINSTALL = 'uninstall',
  UNEXPECTED = 'unexpected',
}

export interface InstallButtonProps {
  loading?: boolean
  disabled?: boolean
  mobile?: boolean
  mode?: ButtonMode
  onClick?: () => void
}

const InstallButton: FC<InstallButtonProps> = ({
  loading = false,
  disabled = false,
  mobile = false,
  mode = ButtonMode.INSTALL,
  onClick,
}) => {
  return loading ? (
    <div
      className={[
        styles.root,
        `${mobile ? styles['skeleton-mobile'] : styles['skeleton-desktop']}`,
      ].join(' ')}
    ></div>
  ) : (
    <button
      type="button"
      data-testid="install-button"
      disabled={disabled}
      className={[
        styles.root,
        `${mobile ? styles.mobile : styles.desktop}`,
        styles[mode],
        `${disabled ? styles.disabled : ''}`,
      ].join(' ')}
      onClick={onClick}
    >
      {mobile ? <SvgIcon icon={mode} /> : mode}
    </button>
  )
}

export default InstallButton
