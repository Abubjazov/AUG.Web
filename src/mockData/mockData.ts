import { IDapplet, ITag } from 'store/slices/dappletsSlice'
import { IList, IUserDapplet } from 'store/slices/userDataSlice'

export const mockUserTags: ITag[] = [
  { tagId: 'darP5Jyz8yirTMsfY9RP', tagName: 'Twitter' },
  { tagId: 'JW3UFtZ5HgATcwldsJ1T', tagName: 'Facebook' },
  { tagId: '6UKxNrMzte6RCWIlssvM', tagName: 'Top 10' },
  { tagId: 'ZbEuko9nOlQYKG6I1Efx', tagName: 'Test' },
  { tagId: '521keF1Adymn8Oo896vv', tagName: 'ToDo' },
]

export const mockUserLists: IList[] = [
  { listId: 'XfLdXIykGHsgIjdJkTaE', listName: 'My special list' },
  { listId: 'c6rClgOltChco2XnyVDp', listName: 'Financial list' },
]

export const mockUserDapplets: IUserDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    userTags: ['JW3UFtZ5HgATcwldsJ1T', '521keF1Adymn8Oo896vv'],
    dappletState: true,
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    userTags: [],
    dappletState: true,
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    userTags: 'N/A',
    dappletState: true,
  },
]

export const mockCommunityTags: ITag[] = [
  { tagId: 'XoEVTHXhfMAjIWsi4wxY', tagName: 'Top 100' },
  { tagId: '3LYUFQTbdNULwBgCkfXx', tagName: 'Finances' },
  { tagId: 'XZ2zPZqa4fPxM9mLA9xR', tagName: 'Media' },
  { tagId: 'C69B5fQ7ccW6Q1SC0yow', tagName: 'Favourites' },
]

export const mockDapplets: IDapplet[] = [
  {
    dappletId: 'ECNk2nNngwGXouvMpjWt',
    appOwner: 'Satoshi Nakamoto',
    circulatingSupply: '19534212',
    communityTags: ['3LYUFQTbdNULwBgCkfXx', 'C69B5fQ7ccW6Q1SC0yow'],
    date: '1230757200',
    fullDesc:
      'Bitcoin is a peer-to-peer online currency, meaning that all transactions happen directly between equal, independent network participants, without the need for any intermediary to permit or facilitate them. Bitcoin was created, according to Nakamoto’s own words, to allow “online payments to be sent directly from one party to another without going through a financial institution.”',
    fullyDilutedMarketCap: '738866583732',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-1.01',
    maxSupply: '21000000',
    name: 'Bitcoin Contracts',
    shortDesc:
      'Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. ',
    shortName: 'BTC',
    totalSupply: '19534106',
    volume: '10.9',
    volumePerMarketCap: '1.58',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHG',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'ErcSJarm6Ck1rzq7yhHGCOPY',
    appOwner: 'Paxos Standard',
    circulatingSupply: '241.792',
    communityTags: [],
    date: '1567285200',
    fullDesc:
      'Pax Gold (PAXG) is a gold-backed cryptocurrency, launched by the creators of Paxos Standard (PAX) in September 2019. As an ERC-20 token operating on the Ethereum blockchain, Pax Gold is tradeable on a large variety of exchanges and has become an accessible way for traders to start investing in gold.',
    fullyDilutedMarketCap: '475547334',
    logo: '/public/images/notAvailable.svg',
    marketCap: '-0.06',
    maxSupply: 'N/A',
    name: 'PAX Gold',
    shortDesc:
      'Charles Cascarilla is the founder and chief executive officer of both Paxos Standard and Pax Gold.',
    shortName: 'PAXG',
    totalSupply: '241.792',
    volume: '43.3',
    volumePerMarketCap: '1.49',
  },
  {
    dappletId: 'T6hUx4HWCKtnIEfwQxYp',
    appOwner: 'XRP Ledger',
    circulatingSupply: '53615837759',
    communityTags: 'N/A',
    date: '1620939600',
    fullDesc:
      'Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.',
    fullyDilutedMarketCap: '65183420834',
    logo: '/public/images/notAvailable.svg',
    marketCap: '5.66',
    maxSupply: '100000000000',
    name: 'XRP Ledger',
    shortDesc:
      'Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.',
    shortName: 'XRP',
    totalSupply: '99988316618',
    volume: '132.26',
    volumePerMarketCap: '5.37',
  },
  {
    dappletId: 'QbWG3sKvfgFcP5RtskMp',
    appOwner: 'Realcoin',
    circulatingSupply: '85367759449',
    communityTags: [],
    date: '1388523600',
    fullDesc:
      'Launched in 2014, Tether is a blockchain-enabled platform designed to facilitate the use of fiat currencies in a digital manner. Tether works to disrupt the conventional financial system via a more modern approach to money. Tether has made headway by giving customers the ability to transact with traditional currencies across the blockchain, without the inherent volatility and complexity typically associated with a digital currency. As the first blockchain-enabled platform to facilitate the digital use of traditional currencies (a familiar, stable accounting unit), Tether has democratised cross-border transactions across the blockchain.',
    fullyDilutedMarketCap: '88708606718',
    logo: '/dappletsLogos/tether.png',
    marketCap: '0.01',
    maxSupply: 'N/A',
    name: 'Tether USDt',
    shortDesc:
      'Tether works to disrupt the conventional financial system via a more modern approach to money.',
    shortName: 'USDT',
    totalSupply: '88623656724',
    volume: '41.44',
    volumePerMarketCap: '32.16',
  },
]
