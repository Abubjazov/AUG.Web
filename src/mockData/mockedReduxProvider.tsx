import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
