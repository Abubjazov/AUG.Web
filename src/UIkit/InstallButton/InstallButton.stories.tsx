import type { Meta, StoryObj } from '@storybook/react'

import '/src/index.css'

import InstallButton, {
  InstallButtonProps,
  InstallButtonMode,
} from './InstallButton'

const meta: Meta<InstallButtonProps> = {
  component: InstallButton,
  title: 'UIkit/Install Button',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - install button',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    mode: {
      description: 'Button appearance',
      defaultValue: InstallButtonMode.INSTALL,
    },
    mobile: {
      description: 'Button appearance for mobile version (icon button)',
    },
    disabled: {
      description: 'Button is not active',
    },
    loading: {
      description: 'Appearance of the button when loading content (skeleton)',
    },
    onClick: {
      description: 'Optional click handler',
    },
  },
}

export default meta

type Story = StoryObj<InstallButtonProps>

export const Default: Story = {}
