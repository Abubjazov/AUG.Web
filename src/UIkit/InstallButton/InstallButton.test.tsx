import { screen, fireEvent, render } from '@testing-library/react'
import { mockedReduxProvider as Provider } from 'mockData/mockedReduxProvider'

import InstallButton from './InstallButton'

describe('InstallButton', () => {
  test('should render InstallButton with INSTALL mode', () => {
    const { asFragment } = render(
      <Provider>
        <InstallButton dappletId={4} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render InstallButton with INSTALLED mode', () => {
    const { asFragment } = render(
      <Provider>
        <InstallButton dappletId={1} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render InstallButton with skeleton', () => {
    const { asFragment } = render(
      <Provider>
        <InstallButton dappletId={1} loading />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render InstallButton with UNINSTALL mode', () => {
    const { asFragment } = render(
      <Provider>
        <InstallButton dappletId={1} />
      </Provider>,
    )

    fireEvent.mouseEnter(screen.getByTestId('install-button'))

    expect(asFragment()).toMatchSnapshot()
  })
})
