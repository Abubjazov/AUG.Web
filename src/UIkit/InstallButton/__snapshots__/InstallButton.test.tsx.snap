// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstallButton > should call installDapplet function on button click, on INSTALL mode 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet QbWG3sKvfgFcP5RtskMp"
    class="_root_cc4fb1 _desktop_cc4fb1 _install_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    install
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should call installDapplet function on button click, on UNISTALL mode 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _uninstall_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    uninstall
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with "disabled" 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 _disabled_cc4fb1"
    data-testid="install-button"
    disabled=""
    type="button"
  >
    installed
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with "loading" 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    <svg
      class="small-spinner"
      height="20"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="20"
    >
      <path
        d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
        fill="none"
        stroke="#fff"
        stroke-dasharray="42.76482137044271 42.76482137044271"
        stroke-linecap="round"
        stroke-width="8"
      >
        <animate
          attributeName="stroke-dashoffset"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          values="0;256.58892822265625"
        />
      </path>
    </svg>
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with "loading" mobile 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _mobile_cc4fb1 _installed_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    <svg
      class="small-spinner"
      height="20"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="20"
    >
      <path
        d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
        fill="none"
        stroke="#fff"
        stroke-dasharray="42.76482137044271 42.76482137044271"
        stroke-linecap="round"
        stroke-width="8"
      >
        <animate
          attributeName="stroke-dashoffset"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          values="0;256.58892822265625"
        />
      </path>
    </svg>
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with INSTALL mode 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet QbWG3sKvfgFcP5RtskMp"
    class="_root_cc4fb1 _desktop_cc4fb1 _install_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    install
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with INSTALL mode mobile 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet QbWG3sKvfgFcP5RtskMp"
    class="_root_cc4fb1 _mobile_cc4fb1 _install_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    <svg
      fill="none"
      height="14"
      viewBox="0 0 14 14"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip0_36_2615)"
      >
        <path
          d="M13.3002 9.09998V11.9C13.3002 12.2713 13.1527 12.6274 12.8901 12.8899C12.6276 13.1525 12.2715 13.3 11.9002 13.3H2.1002C1.72889 13.3 1.3728 13.1525 1.11025 12.8899C0.847695 12.6274 0.700195 12.2713 0.700195 11.9V9.09998"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M3.5 5.59998L7 9.09998L10.5 5.59998"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M7 9.09995V0.699951"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </g>
      <defs>
        <clippath
          id="clip0_36_2615"
        >
          <rect
            fill="white"
            height="14"
            width="14"
          />
        </clippath>
      </defs>
    </svg>
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with INSTALL mode mobile setMode = EInstallButtonMode.DISPLAY_NONE 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet QbWG3sKvfgFcP5RtskMp"
    class="_root_cc4fb1 _mobile_cc4fb1 _display-none_cc4fb1 "
    data-testid="install-button"
    type="button"
  />
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with INSTALLED mode 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    installed
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with UNINSTALL mode 1`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _uninstall_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    uninstall
  </button>
</DocumentFragment>
`;

exports[`InstallButton > should render InstallButton with UNINSTALL mode 2`] = `
<DocumentFragment>
  <button
    aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
    class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
    data-testid="install-button"
    type="button"
  >
    installed
  </button>
</DocumentFragment>
`;
